Sources/FoundationInternationalization/ICU/ICU+StringConverter.swift

```diff
@@ -0,0 +1,202 @@
+                return data.withUnsafeBytes { src in
+                    ucnv_toUChars(
+                        converter,
+                        dest,
+                        capacity,
+                        src.baseAddress,
+                        srcLength,
+                        &status
+                    )
+                }
+            }
+        }
+    }
+
+    func encode(string: String, allowLossyConversion lossy: Bool) -> Data?  {
+        return _converter.withLock { (converter) -> Data? in
+            defer {
+                ucnv_resetFromUnicode(converter)
+            }
+
+            let utf16Rep = string.utf16
+            let uchars = UnsafeMutableBufferPointer<UChar>.allocate(capacity: utf16Rep.count)
+            _ = uchars.initialize(fromContentsOf: utf16Rep)
+            defer {
+                uchars.deallocate()
+            }
+
+            let srcLength = uchars.count
+            let capacity = srcLength * Int(ucnv_getMaxCharSize(converter)) + 1
+            let dest = UnsafeMutableRawPointer.allocate(
+                byteCount: capacity,
+                alignment: MemoryLayout<CChar>.alignment
+            )
+
+            var status: UErrorCode = U_ZERO_ERROR
+            if lossy {
+                var lossyChar: UChar = encoding == .ascii ? 0xFF : 0x3F
+                ucnv_setSubstString(
+                    converter,
+                    &lossyChar,
+                    1,
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+
+                ucnv_setFromUCallBack(
+                    converter,
+                    UCNV_FROM_U_CALLBACK_SUBSTITUTE,
+                    nil, // newContext
+                    nil, // oldAction
+                    nil, // oldContext
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+            } else {
+                ucnv_setFromUCallBack(
+                    converter,
+                    UCNV_FROM_U_CALLBACK_STOP,
+                    nil, // newContext
+                    nil, // oldAction
+                    nil, // oldContext
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+            }
+
+            let actualLength = ucnv_fromUChars(
+                converter,
+                dest,
+                CInt(capacity),
+                uchars.baseAddress,
+                CInt(srcLength),
+                &status
+            )
+            guard status.isSuccess else { return nil }
+            return Data(
+                bytesNoCopy: dest,
+                count: Int(actualLength),
+                deallocator: .custom({ pointer, _ in pointer.deallocate() })
+            )
+        }
+    }
+}
+
+extension ICU.StringConverter {
+    nonisolated(unsafe) static private var _converters: LockedState<[String.Encoding: ICU.StringConverter]> = .init(initialState: [:])
```

Comment:
I think this should just be a `static private let` (the immutable-ness affects the `LockedState` itself but not the contents IIRC, and this isn't unsafe because it's guarded by a lock)

Reply:
Indeed it should be.
This is also a vestige of my first impl, I guess.